api/envoy/api/v2/auth/cert.proto

```diff
@@ -171,6 +171,19 @@ message TlsSessionTicketKeys {
 }
 
 message CertificateValidationContext {
+
+  // Peer certificate verification mode.
+  enum TrustChainVerification {
+    // Perform default certificate verification (eg, against CA / verification lists)
+    VERIFY_TRUST_CHAIN = 0;
+    // Certificates that fail verification will be permitted and their details hoisted in the
+    // x-forwarded-untrusted-client-cert HTTP Header.
```

Comment:
Nit: `// *x-forwarded-untrusted-client-cert* HTTP header`. I.e. use RST formatting for header names, s/Header/header/g below as well.